service/src/main/resources/api/service_openapi.yaml

```diff
@@ -994,6 +1093,129 @@ components:
       items:
         type: string
 
+    CohortV2:
+      type: object
+      properties:
+        id:
+          description: ID of the cohort, immutable
+          type: string
+        underlayName:
+          $ref: "#/components/schemas/UnderlayNameV2"
+        displayName:
+          $ref: "#/components/schemas/CohortDisplayNameV2"
+        description:
+          $ref: "#/components/schemas/CohortDescriptionV2"
+        criteriaGroups:
+          type: array
+          description: Groups of criteria that define the cohort
+          items:
+            $ref: "#/components/schemas/CriteriaGroupV2"
+        filter:
+          # TODO: Remove this filter property once we move the code that converts the criteriaGroups to a filter
+          # from the UI to the backend.
+          $ref: "#/components/schemas/FilterV2"
+
+    CohortListV2:
+      type: array
+      items:
+        $ref: "#/components/schemas/CohortV2"
+
+    CohortCreateInfoV2:
+      type: object
+      properties:
+        underlayName:
```

Comment:
I think specifying which fields users can update or specify on creation is part of the API definition. Alternative is to reuse the same object, but then list in the description which fields are updatable, and that seems more error-prone.

Reply:
Per offline conversation, we decided to leave the API as is for now -- with the update/create fields explicitly specified in the OpenAPI spec rather than enforced by Java code on the backend. We may reevaluate other approaches in the future.